autogpt_platform/backend/backend/blocks/llm.py

```diff
@@ -104,6 +108,11 @@ class Input(BlockSchema):
         prompt_values: dict[str, str] = SchemaField(
             advanced=False, default={}, description="Values used to fill in the prompt."
         )
+        ollama_host: str = SchemaField(
```

Comment:
@Bentlybro is there a way to conditionally show stuff like this if ollama is the selected model:

Reply:
@ntindle i dont think we have that setup yet but maybe that is something we should maybe look into getting setup?

Reply:
Just thinking, would it be worth splitting model selection into provider then model, rather than just model? Some providers have a wide variety of models (i.e. Ollama) which may overlap with other providers, so being able to choose the model and provider would let users have more control. It'd also make it a tad easier to have conditional inputs on the blocks I imagine as we wouldn't have to look up provider based on the model, it'd be in the block inputs.

Let me know and I can try and get a pr out for that functionality soon

Reply:
I think that's a good very good idea, it makes a lot more sense being able to do it that way, @ntindle @Torantulino what do we think? it should be pretty easy to do and it should just be changes to the block its self as far as I can see.

Reply:
@Bentlybro the one issue I see with it is getting the frontend to display that properly - we'd need to make changes to the JSON schema which gets passed to the frontend to render the available model selection. Or we could just throw an error when model and provider don't match, but that seems pretty hostile to new users who might not understand the differences between the providers and models.

Reply:
@Bentlybro @ntindle 
Going to resolve this so we can go ahead and merge once i finish rebasing for the author, will open an issue for the above and tag you